drivers/rtc/rtc_rv8263.c

```diff
@@ -422,86 +422,69 @@ static int rv8263c8_alarm_set_time(const struct device *dev, uint16_t id, uint16
 		return -EINVAL;
 	}
 
-	/* Disable the alarm when mask is zero. */
-	if (mask == 0) {
-		return i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
-					      RV8263C8_BM_ALARM_INT_ENABLE,
-					      RV8263C8_BM_ALARM_INT_DISABLE);
-	}
-
 	if (!rtc_utils_validate_rtc_time(timeptr, mask)) {
 		LOG_ERR("Invalid mask!");
 		return -EINVAL;
 	}
 
-	regs[0] = bin2bcd(timeptr->tm_sec) & SECONDS_BITS;
-	regs[1] = bin2bcd(timeptr->tm_min) & MINUTES_BITS;
-	regs[2] = bin2bcd(timeptr->tm_hour) & HOURS_BITS;
-	regs[3] = bin2bcd(timeptr->tm_mday) & DATE_BITS;
-	regs[4] = bin2bcd(timeptr->tm_wday) & WEEKDAY_BITS;
-
-	if (mask & RTC_ALARM_TIME_MASK_SECOND) {
-		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_SECONDS_ALARM,
-					    RV8263C8_BM_ALARM_ENABLE | regs[0]);
+	if (mask == 0) {
+		err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
+					     RV8263C8_BM_ALARM_INT_ENABLE,
+					     RV8263C8_BM_ALARM_INT_DISABLE);
```

Comment:
```suggestion
	if (mask == 0) {
		err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
					     RV8263C8_BM_ALARM_INT_ENABLE | RV8263C8_BM_AF,
					     RV8263C8_BM_ALARM_INT_DISABLE);
```
I would also clear the AF flag here.